### PR TITLE
18EU Initial Re-commit

### PR DESCRIPTION
**Pull request by ryandriskel**
From: https://github.com/tobymao/18xx/issues/792

This PR adds an initial implementation of 18EU with the auction and first operating rounds of minors supported.

### Changes in common code
**round/auction**: The input and place bid inputs were not coded into the view for minors, since 18Mag is a draft. They have been added. I also moved the lambda function into its own distinct function for readability. There is probably some refactoring that could be done here to reduce duplication of code, but I wanted to save that for its own PR.

**Auctioner / Passable Bid Steps**: It was necessary to add `|| bid.minor` to make the auction code generically support selecting minors. I wanted to eliminate having to worry about this in future auctions and thus created the `bid_target` function.

### Change to 18Mag
18Mag was the only other title implementing minors in the initial auction. However, 18Mag is a draft, thus the input and place bid inputs were not coded into the view at all. It was necessary to add the _auctioneer_ function as a way to indicate to the view that it should not show the newly supported inputs.

Screen Shot 2021-02-11 at 4 31 56 PM

Screen Shot 2021-02-11 at 4 30 26 PM

Screen Shot 2021-02-11 at 4 30 52 PM

Screen Shot 2021-02-11 at 4 29 25 PM

### CODE REVIEW

**Comment by g3lutz** [comment]
Okay I am in.
I will read some more this afternoon and this weekend.

File: lib/engine/step/g_18_eu/modified_dutch_auction.rb [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+          next_entity!
+        end
+
+        def resolve_bids
+          return unless @auctioning
+
+          target = @auctioning
+
+          return unless @active_bidders.one?
+          return unless @bids[@auctioning].any?
```
Comment (tobymao) [inline lib/engine/step/g_18_eu/modified_dutch_auction.rb:207]: empty

File: lib/engine/step/g_18_eu/modified_dutch_auction.rb [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        end
+
+        def starting_bid(_entity)
+          @base_starting_bid - @current_reduction
+        end
+
+        # TODO: Decrease minimum bid if all passed
+        def min_bid(entity)
+          return unless entity
+          return starting_bid(entity) unless @bids[entity].any?
```
Comment (tobymao) [inline lib/engine/step/g_18_eu/modified_dutch_auction.rb:99]: empty

**Comment by tobymao** [comment]
svgs kinda big can they be compressed?

**Comment by ryandriskel** [comment]
> svgs kinda big can they be compressed?

I have run them through https://vecta.io/nano ... there are still a couple larger ones due to complexity. These are the original complex logos and _could_ be simplified if necessary.

**Pull request merged**